deploy/helm/templates/configMap.yaml

```
@@ -7,7 +11,17 @@ metadata:
     {{- include "appsmith.labels" . | nindent 4 }}
 data:
   {{- range $key, $value := .Values.applicationConfig }}
+  {{- if and (eq "APPSMITH_MONGODB_URI" $key) ( not $value) }}
+  {{- if $.Values.mongodb.enabled }}
+  {{ $key }}: mongodb+srv://{{ $mongoUser }}:{{ $mongoPassword }}@{{ $mongoServicename }}.{{ $nameSpace }}.svc.cluster.local/appsmith?retryWrites=true&authSource=admin&ssl=false
+  {{- end }}
+  {{- end }}
```

This looks like we're missing a condition. If condition on line 14 is true, and on line 15 is false, that is `mongodb.enabled` is `false`, we should just revert to previous behavior, right? Like this:

```suggestion
  {{- if and (eq "APPSMITH_MONGODB_URI" $key) ( not $value) }}
  {{- if $.Values.mongodb.enabled }}
  {{ $key }}: mongodb+srv://{{ $mongoUser }}:{{ $mongoPassword }}@{{ $mongoServicename }}.{{ $nameSpace }}.svc.cluster.local/appsmith?retryWrites=true&authSource=admin&ssl=false
  {{- else }}
  {{ $key }}: {{ $value | quote }}
  {{- end }}
  {{- end }}
```

This is to keep existing behavior, more than anything else. What say? Similarly for Redis as well.

---

Yes, previous behavior is an empty value/whatever the user gives right. 
So,
if the `APPSMITH_MONGODB_URI` doesnt contain the value and if mongodb is enabled.
```{{ $key }}: mongodb+srv://{{ $mongoUser }}:{{ $mongoPassword }}@{{ $mongoServicename }}.{{ $nameSpace }}.svc.cluster.local/appsmith?retryWrites=true&authSource=admin&ssl=false```
if the `APPSMITH_MONGODB_URI` doesnt contain the value and if mongodb is disabled. 
 ```{{ $key }}: {{ $value | quote }}```

---

That will happen in the end because of the for loop at line 25.

---

Oh, I see. But, that means `APPSMITH_MONGODB_URI` and `APPSMITH_REDIS_URL` could be duplicated, right? Like, if they are printed from lines 16 and 21, they can also be printed in the for loop no? Can we rewrite this using `else` statements to avoid duplicates?

---

You mean what if we have duplicate values in values.yaml. Like, having two different values for APPSMITH_MONGODB_URI in the values.yaml ?